misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -370,10 +375,35 @@ private static String getMappingFromServer(JsonArray dataToSend,
                 }
             }
             JsonParser parser = new JsonParser();
-            return parser.parse(response.getData()).getAsJsonObject().get("answer").getAsString();
+            return parser.parse(response.getData()).getAsJsonObject().get("answer").toString();
         } catch (IOException e) {
             throw new IOException("Error connecting the AI service" + e.getMessage(), e);
         }
     }
-}
 
+    private static String generateMappingFunction(String mappingFromServer, String foundTypeLeft,
+                                                  String foundTypeRight, String leftModule, String rightModule) {
+
+        String leftType = foundTypeLeft;
+        String rightType = foundTypeRight;
+
+        mappingFromServer = mappingFromServer.replaceAll("\"", "");
+        mappingFromServer = mappingFromServer.replaceAll(",", ", ");
+        mappingFromServer = mappingFromServer.replaceAll(":", ": ");
+
+        if (leftModule != null) {
```

Comment:
have you written unit tests to check if the mapping function is generated correctly?
if so how many unit tests?

Reply:
Yes. In 15 tests, I check if the mapping function is generated correctly.